packages/modules/src/modules/banners/choiceCardsBanner/ChoiceCardsBanner.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import { Container, Columns, Column, Inline } from '@guardian/src-layout';
+import { Button } from '@guardian/src-button';
+import { SvgRoundelDefault } from '@guardian/src-brand';
+import { SvgCross } from '@guardian/src-icons';
+import { BannerText } from '../common/BannerText';
+import { BannerId, BannerRenderProps } from '../common/types';
+import {
+    banner,
+    closeButtonStyles,
+    containerOverrides,
+    copyColumn,
+    heading,
+    highlightedTextBlueBanner,
+    highlightedTextYellowBanner,
+    iconAndClosePosition,
+    choiceCardsColumn,
+    logoContainer,
+    paragraph,
+    columnMarginOverrides,
+} from './choiceCardsBannerStyles';
+import { createInsertEventFromTracking, getLocalCurrencySymbol } from '@sdc/shared/dist/lib';
+import { createViewEventFromTracking } from '@sdc/shared/dist/lib';
+import { ChoiceCards } from './components/ChoiceCards';
+import { ContributionFrequency } from '@sdc/shared/src/types';
+import { HasBeenSeen, useHasBeenSeen } from '../../../hooks/useHasBeenSeen';
+
+type ButtonPropTypes = {
+    onClick: (evt: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    bannerId: string;
+};
+
+const CloseButton = (props: ButtonPropTypes): ReactElement => {
+    const closeComponentId = `${props.bannerId} : close`;
+    return (
+        <Button
+            css={closeButtonStyles}
+            data-link-name={closeComponentId}
+            onClick={props.onClick}
+            icon={<SvgCross />}
+            size="small"
+            priority="tertiary"
+            hideLabel
+        >
+            Close this banner
+        </Button>
+    );
+};
+
+export interface ChoiceCardSelection {
+    frequency: ContributionFrequency;
+    amount: number | 'other';
+}
+
+export type ChoiceCardsBannerRenderProps = {
+    bannerId: BannerId;
+    backgroundColor: string;
+    headingColor: string;
+};
+
+export const ChoiceCardsBanner = ({
+    bannerId,
+    backgroundColor,
+    headingColor,
+    onCloseClick,
+    content,
+    choiceCardAmounts,
+    countryCode,
+    submitComponentEvent,
+    tracking,
+    numArticles,
+}: Omit<
+    BannerRenderProps,
+    'onCtaClick' | 'onSecondaryCtaClick' | 'onNotNowClick' | 'reminderTracking'
+> &
+    ChoiceCardsBannerRenderProps): JSX.Element => {
+    const [choiceCardSelection, setChoiceCardSelection] = useState<
+        ChoiceCardSelection | undefined
+    >();
+    const [hasBeenSeen, setNode] = useHasBeenSeen({ threshold: 0 }, true) as HasBeenSeen;
+
+    useEffect(() => {
+        if (hasBeenSeen && tracking) {
```

Comment:
no sure how Ophan tracking works or how to check this. Maybe if get some time.

Reply:
Most of these functions come from tracking.ts

Reply:
Another time, when there is more time perhaps I could get a refresher on this. Will look at tracking.ts